algorithms/src/Kokkos_Random.hpp

```diff
@@ -1518,6 +1531,302 @@ void fill_random(const ExecutionSpace& exec, ViewType a, RandomPool g,
+      if (idx < static_cast<IndexType>(a.extent(0))) {
+        for (IndexType k = 0; k < static_cast<IndexType>(a.extent(1)); k++)
+          for (IndexType l = 0; l < static_cast<IndexType>(a.extent(2)); l++)
+            for (IndexType m = 0; m < static_cast<IndexType>(a.extent(3)); m++)
+              for (IndexType n = 0; n < static_cast<IndexType>(a.extent(4));
+                   n++)
+                for (IndexType o = 0; o < static_cast<IndexType>(a.extent(5));
+                     o++)
+                  for (IndexType p = 0; p < static_cast<IndexType>(a.extent(6));
+                       p++)
+                    a(idx, k, l, m, n, o, p) =
+                        Rand::draw_normal(gen, mean, std_dev);
+      }
+    }
+    rand_pool.free_state(gen);
+  }
+};
+
+template <class ViewType, class RandomPool, int loops, class IndexType>
+struct fill_random_functor_normal<ViewType, RandomPool, loops, 8, IndexType> {
+  ViewType a;
+  RandomPool rand_pool;
+  typename ViewType::const_value_type mean, std_dev;
+
+  using Rand = rand<typename RandomPool::generator_type,
+                    typename ViewType::non_const_value_type>;
+
+  fill_random_functor_normal(ViewType a_, RandomPool rand_pool_,
+                             typename ViewType::const_value_type mean_,
+                             typename ViewType::const_value_type std_dev_)
+      : a(a_), rand_pool(rand_pool_), mean(mean_), std_dev(std_dev_) {}
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(IndexType i) const {
+    typename RandomPool::generator_type gen = rand_pool.get_state();
+    for (IndexType j = 0; j < loops; j++) {
+      const IndexType idx = i * loops + j;
+      if (idx < static_cast<IndexType>(a.extent(0))) {
+        for (IndexType k = 0; k < static_cast<IndexType>(a.extent(1)); k++)
+          for (IndexType l = 0; l < static_cast<IndexType>(a.extent(2)); l++)
+            for (IndexType m = 0; m < static_cast<IndexType>(a.extent(3)); m++)
+              for (IndexType n = 0; n < static_cast<IndexType>(a.extent(4));
+                   n++)
+                for (IndexType o = 0; o < static_cast<IndexType>(a.extent(5));
+                     o++)
+                  for (IndexType p = 0; p < static_cast<IndexType>(a.extent(6));
+                       p++)
+                    for (IndexType q = 0;
+                         q < static_cast<IndexType>(a.extent(7)); q++)
+                      a(idx, k, l, m, n, o, p, q) =
+                          Rand::draw_normal(gen, mean, std_dev);
+      }
+    }
+    rand_pool.free_state(gen);
+  }
+};
+
+template <class ExecutionSpace, class ViewType, class RandomPool,
+          class IndexType = int64_t>
+void fill_random_normal(const ExecutionSpace& exec, ViewType a, RandomPool g,
+                        typename ViewType::const_value_type mean,
+                        typename ViewType::const_value_type std_dev) {
+  int64_t LDA = a.extent(0);
+  if (LDA > 0)
+    parallel_for(
+        "Kokkos::fill_random",
```

Comment:
the label here is missing "normal"

Reply:
@fnrizzi Thanks for catching this. I added "normal".